src/main/java/com/puppycrawl/tools/checkstyle/checks/design/DesignForExtensionCheck.java

```diff
@@ -322,7 +348,36 @@ private static boolean branchContainsJavadocComment(DetailAST token) {
             }
             curNode = toVisit;
         }
+        if (curNode != null) {
+            result = isValidJavadocComment(curNode, requiredJavadocPhrase);
+        }
+        return result;
+    }
 
+    /**
+     * Checks whether a javadoc follows the specified comment pattern.
+     *
+     * @param token tree token
+     * @param requiredJavadocPhrase required javadoc comment pattern
+     * @return true if the javadoc comment of the specified token follows the given pattern
+     */
+    private static boolean isValidJavadocComment(DetailAST token, Pattern requiredJavadocPhrase) {
+        final Pattern javadocStars = Pattern.compile("\\*\\s*");
+        final Pattern javadocWhiteSpaces = Pattern.compile("\\s+");
+        String javadoc = javadocWhiteSpaces
+                .matcher(javadocStars.matcher(JavadocUtil.getBlockCommentContent(token))
+                .replaceAll(""))
+                .replaceAll(" ");
+
+        final int firstTagIndex = javadoc.indexOf('@');
+        if (firstTagIndex != -1) {
+            javadoc = javadoc.substring(0, firstTagIndex);
+        }
+        final Matcher matcher = requiredJavadocPhrase.matcher(javadoc.trim());
+        boolean result = false;
```

Comment:
`return matcher.find();` ?

Reply:
Sorry, my bad, I had additional conditions before, I'll amend it in the next commit.